src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnusedLocalVariableCheck.java

```diff
@@ -0,0 +1,449 @@
+                    || parent.getFirstChild() == ast
+                    && parent.getLastChild().getType() != TokenTypes.LITERAL_NEW) {
+                result = true;
+            }
+        }
+        else {
+            result = !TokenUtil.findFirstTokenByPredicate(ast,
+                            childAst -> {
+                                return TokenUtil.isOfType(childAst,
+                                        UNACCEPTABLE_CHILD_OF_DOT);
+                            })
+                    .isPresent();
+        }
+        return result;
+    }
+
+    /**
+     * Checks the identifier ast.
+     *
+     * @param identAst DetailAst of type {@link TokenTypes#IDENT}
+     */
+    private void checkIdentifierAst(DetailAST identAst) {
+        for (VariableDesc variableDesc : variables) {
+            if (identAst.getText().equals(variableDesc.getName())
+                    && !isLeftHandSideValue(identAst)) {
+                variableDesc.registerAsUsed();
+                break;
+            }
+        }
+    }
+
+    /**
+     * Find the scope of variable.
+     *
+     * @param variableDef DetailAst of type {@link TokenTypes#VARIABLE_DEF}
+     * @return scope of variableDef
+     */
+    private static DetailAST findScopeOfVariable(DetailAST variableDef) {
+        final DetailAST result;
+        final DetailAST parentAst = variableDef.getParent();
+        if (parentAst.getType() == TokenTypes.SLIST) {
+            result = parentAst;
+        }
+        else {
+            final DetailAST grandParent = parentAst.getParent();
+            final DetailAST scopeAst = grandParent.findFirstToken(TokenTypes.SLIST);
+            if (scopeAst == null) {
+                result = grandParent;
+            }
+            else {
+                result = scopeAst;
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Checks whether the DetailAst of type {@link TokenTypes#IDENT} is
+     * used as left-hand side value. An identifier is being used as a left-hand side
+     * value if it is used as the left operand of an assignment or as an
+     * operand of a stand-alone increment
+     *
+     * @param identAst DetailAst of type {@link TokenTypes#IDENT}
+     * @return True if identAst is used as a left-hand side value otherwise false.
+     */
+    private static boolean isLeftHandSideValue(DetailAST identAst) {
+        final DetailAST parent = identAst.getParent();
+        return isStandAloneIncrement(identAst)
+                || parent.getType() == TokenTypes.ASSIGN
+                && identAst != parent.getLastChild();
+    }
+
+    /**
+     * Checks whether the DetailAst of type {@link TokenTypes#IDENT} is used as
+     * an operand of a stand-alone increment.
+     *
+     * @param identAst DetailAst of type {@link TokenTypes#IDENT}
+     * @return True if identAst is used as an operand of stand-alone increment otherwise false
+     */
+    private static boolean isStandAloneIncrement(DetailAST identAst) {
```

Comment:
name is a bit confusing. It is not only increment check, but also decrement and some others

Reply:
done, renamed to `isStandAloneIncrementOrDecrement(..)`
>...and some others

The other tokens are present around the inc and dec operator so the method name shouldn't include them IMO